// @storybook/addon-jest/dist/hoc/provideJestResult.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var provideTests = function provideTests(Component) {
  var _class, _temp;

  return _temp = _class =
  /*#__PURE__*/
  function (_React$Component) {
    (0, _inherits2.default)(TestProvider, _React$Component);

    function TestProvider() {
      var _getPrototypeOf2;

      var _this;

      (0, _classCallCheck2.default)(this, TestProvider);

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(TestProvider)).call.apply(_getPrototypeOf2, [this].concat(args)));
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {});
      (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "onAddTests", function (_ref) {
        var kind = _ref.kind,
            storyName = _ref.storyName,
            tests = _ref.tests;

        _this.setState({
          kind: kind,
          storyName: storyName,
          tests: tests
        });
      });
      return _this;
    }

    (0, _createClass2.default)(TestProvider, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        var _this2 = this;

        var _this$props = this.props,
            channel = _this$props.channel,
            api = _this$props.api;
        this.stopListeningOnStory = api.onStory(function () {
          _this2.onAddTests({});
        });
        channel.on('storybook/tests/add_tests', this.onAddTests);
      }
    }, {
      key: "componentWillUnmount",
      value: function componentWillUnmount() {
        var channel = this.props.channel;

        if (this.stopListeningOnStory) {
          this.stopListeningOnStory();
        }

        channel.removeListener('storybook/tests/add_tests', this.onAddTests);
      }
    }, {
      key: "render",
      value: function render() {
        var active = this.props.active;
        return active ? _react.default.createElement(Component, this.state) : null;
      }
    }]);
    return TestProvider;
  }(_react.default.Component), (0, _defineProperty2.default)(_class, "propTypes", {
    channel: _propTypes.default.shape({
      on: _propTypes.default.func,
      removeListener: _propTypes.default.func
    }).isRequired,
    api: _propTypes.default.shape({
      onStory: _propTypes.default.func
    }).isRequired,
    active: _propTypes.default.bool
  }), (0, _defineProperty2.default)(_class, "defaultProps", {
    active: true
  }), _temp;
};

var _default = provideTests;
exports.default = _default;
