// @storybook/addon-knobs/dist/components/types/Select.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _components = require("@storybook/components");

var SelectType =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(SelectType, _Component);

  function SelectType() {
    (0, _classCallCheck2.default)(this, SelectType);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SelectType).apply(this, arguments));
  }

  (0, _createClass2.default)(SelectType, [{
    key: "renderOptionList",
    value: function renderOptionList(_ref) {
      var _this = this;

      var options = _ref.options;

      if (Array.isArray(options)) {
        return options.map(function (val) {
          return _this.renderOption(val, val);
        });
      }

      return Object.keys(options).map(function (key) {
        return _this.renderOption(key, options[key]);
      });
    }
  }, {
    key: "renderOption",
    value: function renderOption(key, value) {
      var opts = {
        key: key,
        value: value
      };
      return _react.default.createElement("option", opts, key);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          knob = _this$props.knob,
          _onChange = _this$props.onChange;
      return _react.default.createElement(_components.Select, {
        value: knob.value,
        onChange: function onChange(e) {
          return _onChange(e.target.value);
        },
        size: "flex"
      }, this.renderOptionList(knob));
    }
  }]);
  return SelectType;
}(_react.Component);

SelectType.defaultProps = {
  knob: {},
  onChange: function onChange(value) {
    return value;
  }
};
SelectType.propTypes = {
  knob: _propTypes.default.shape({
    name: _propTypes.default.string,
    value: _propTypes.default.string,
    options: _propTypes.default.oneOfType([_propTypes.default.array, _propTypes.default.object])
  }),
  onChange: _propTypes.default.func
};

SelectType.serialize = function (value) {
  return value;
};

SelectType.deserialize = function (value) {
  return value;
};

var _default = SelectType;
exports.default = _default;
