// @storybook/addon-jest/dist/components/Indicator.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var Indicator =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "ebptffi0"
})(function (_ref) {
  var color = _ref.color,
      size = _ref.size;
  return {
    boxSizing: 'border-box',
    padding: "0 ".concat(size / 2, "px"),
    minWidth: size,
    minHeight: size,
    fontSize: size / 1.4,
    lineHeight: "".concat(size, "px"),
    color: 'white',
    textTransform: 'uppercase',
    borderRadius: size / 2,
    backgroundColor: color
  };
}, function (_ref2) {
  var styles = _ref2.styles;
  return (0, _objectSpread2.default)({}, styles);
});
Indicator.defaultProps = {
  right: false,
  children: ''
};
Indicator.propTypes = {
  color: _propTypes.default.string.isRequired,
  size: _propTypes.default.number.isRequired,
  children: _propTypes.default.node,
  right: _propTypes.default.bool
};
var _default = Indicator;
exports.default = _default;
