// @storybook/addon-knobs/dist/components/types/Date/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var _reactDatetime = _interopRequireDefault(require("react-datetime"));

var _styles = _interopRequireDefault(require("./styles"));

var DateInput =
/*#__PURE__*/
(0, _styled.default)(_reactDatetime.default, {
  target: "e12zg7dq0"
})(_styles.default);

var DateType =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(DateType, _React$Component);

  function DateType() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, DateType);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(DateType)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleChange", function (date) {
      var onChange = _this.props.onChange;
      var value = date.valueOf();
      onChange(value);
    });
    return _this;
  }

  (0, _createClass2.default)(DateType, [{
    key: "shouldComponentUpdate",
    value: function shouldComponentUpdate(nextProps) {
      var knob = this.props.knob;
      return nextProps.knob.value !== knob.value;
    }
  }, {
    key: "render",
    value: function render() {
      var knob = this.props.knob;
      return _react.default.createElement(DateInput, {
        value: knob.value ? new Date(knob.value) : null,
        type: "date",
        onChange: this.handleChange,
        size: "flex"
      });
    }
  }]);
  return DateType;
}(_react.default.Component);

DateType.propTypes = {
  knob: _propTypes.default.shape({
    name: _propTypes.default.string,
    value: _propTypes.default.number
  }).isRequired,
  onChange: _propTypes.default.func.isRequired
};

DateType.serialize = function (value) {
  return String(value);
};

DateType.deserialize = function (value) {
  return parseFloat(value);
};

var _default = DateType;
exports.default = _default;
